frontend/ui/network/manager.lua

```diff
@@ -600,44 +609,29 @@ function NetworkMgr:showNetworkMenu(complete_callback)
                 return
             end
         end
-        -- NOTE: Also supports a disconnect_callback, should we use it for something?
-        --       Tearing down Wi-Fi completely when tapping "disconnect" would feel a bit harsh, though...
-        UIManager:show(require("ui/widget/networksetting"):new{
-            network_list = network_list,
-            connect_callback = complete_callback,
-        })
-    end)
-end
-
-function NetworkMgr:reconnectOrShowNetworkMenu(complete_callback)
-    local info = InfoMessage:new{text = _("Scanning for networks…")}
-    UIManager:show(info)
-    UIManager:nextTick(function()
-        local network_list, err = self:getNetworkList()
-        UIManager:close(info)
-        if network_list == nil then
-            UIManager:show(InfoMessage:new{text = err})
-            return
-        end
         table.sort(network_list,
            function(l, r) return l.signal_quality > r.signal_quality end)
+
         local success = false
-        for dummy, network in ipairs(network_list) do
-            if network.password then
-                success = NetworkMgr:authenticateNetwork(network)
-                if success then
-                    NetworkMgr:obtainIP()
-                    if complete_callback then
-                        complete_callback()
+        if not self.wifi_toggle_long_press then
```

Comment:
When here, are we sure we have always gone thru where you set it to true/false ? No risk to have this set because leaking from a previous (minutes ago) set to true/false ? No other place it would feel safe to reset it ? May be around here after it has been used?

Reply:
I was going to make the same comment as you did above, and, indeed, various other things can call `turnWifiOn` (which is the main caller for this function).